File: 1364-tuple-with-same-product/tuple-with-same-product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        prods = defaultdict(int)
        ans = 0
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                prod = nums[i] * nums[j]
                ans += prods[prod]
                prods[prod] += 1
        return ans * 8

File: 1364-tuple-with-same-product/tuple-with-same-product.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var tupleSameProduct = function (nums) {
    let map = new Map()
    let ans = 0
    for (let i = 0; i < nums.length - 1; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            let prod = nums[i] * nums[j]
            ans += (map.get(prod) || 0)
            map.set(prod, (map.get(prod) || 0) + 1)
        }
    }
    return ans * 8
};